### DISCUSSION TITLE
plpsql function - Select from auth.users

### DISCUSSION THREAD

**Discussion by SteveChurch** [supabase/supabase#12453] [Questions] [answered]
Should a plpsql function be able to `select` from `auth.users`?

<img width="1010" alt="image" src="https://user-images.githubusercontent.com/458008/219214669-036d0819-d070-4a93-bdb4-50460a115e73.png">

I always thought so, but maybe not.

**Comment by GaryAustin1** [comment, marked as answer]
The only way it can is if you make it a "security definer" type, then it will run as the owner (usually postgres).
BUT you really need to know what you are doing and do your own security on the function as anyone can call an rpc and access any data it can as RLS does not apply in this security definer case.

**Reply by SteveChurch** [reply]
:face-palm: - Course, I've managed to get around this by not calling the `auth` schema and instead verifying my needs against a meta table. Thanks.
